doc/unet.txt

```diff
@@ -70,40 +97,46 @@ Code - Citations - Contact
 Code
 ====
 
-The U-Net implementation can be found in the following file:
+The U-Net implementation can be found in the following GitHub repo:
 
-* `unet.py <../code/unet.py>`_ : Main script. Defines the model.
+* `Unet.py <https://github.com/Lasagne/Recipes/blob/master/modelzoo/Unet.py>`_ : Main script. Defines the model
 
 The user can now build a U-Net with a specified number of input channels and number of classes.
 
 First include the Lasagna layer needed to define the U-Net architecture :
 
-.. literalinclude:: ../code/unet.py
+.. literalinclude:: ../code/Unet_lasagne_recipes.py
   :start-after: start-snippet-1
   :end-before: end-snippet-1
 
-Our net variable will be a dictionary containing the layers' name as key and the layer instance as value.
+The *net* variable will be a ordered dictionary containing the layers' name as key and the layer instance as value.
 This is needed to be able to concatenate the feature maps from the contracting to expansive path.
-For example, the first block of conv+conv+block, including the input layer, would be :
 
-.. literalinclude:: ../code/unet.py
-  :start-after: start-snippet-2
-  :end-before: end-snippet-2
 
-Every layer will be added the same way, by specifying its name and
-input layer (or input layers, for concatenation). For example, the last concatenation for this
-architecture takes as input the 'conv1_2' output and the previous layer as follow:
+First the downsampling path :
 
-.. literalinclude:: ../code/unet.py
-  :start-after: start-snippet-3
-  :end-before: end-snippet-3
+.. literalinclude:: ../code/Unet_lasagne_recipes.py
+  :start-after: start-snippet-downsampling
+  :end-before: end-snippet-downsampling
+  
+And then the bottleneck path :  
+  
+.. literalinclude:: ../code/Unet_lasagne_recipes.py
+  :start-after: start-snippet-bottleneck
+  :end-before: end-snippet-bottleneck
+  
+And finally the upsampling path :  
+  
+.. literalinclude:: ../code/Unet_lasagne_recipes.py
+  :start-after: start-snippet-upsampling
+  :end-before: end-snippet-upsampling
 
+And the output path (to obtain *number of classes* feature maps):
 
-To build this network, simply specify the number of input channels and number of classes as follow:
+.. literalinclude:: ../code/Unet_lasagne_recipes.py
+  :start-after: start-snippet-output
+  :end-before: end-snippet-output
 
```

Comment:
Is there a script that loads data, train, and test the model?
I think we should have one for each of the pages.
It would be good to show the difference with the FCN of the previous tutorial, for instance.